### PR TITLE
Improving DPDK capture interface and docs v1

### PR DESCRIPTION
**Pull request by lukashino**
Links to the redmine tickets: 

https://redmine.openinfosecfoundation.org/issues/5857
https://redmine.openinfosecfoundation.org/issues/5858
https://redmine.openinfosecfoundation.org/issues/5859
https://redmine.openinfosecfoundation.org/issues/5937
https://redmine.openinfosecfoundation.org/issues/5936
https://redmine.openinfosecfoundation.org/issues/5925
https://redmine.openinfosecfoundation.org/issues/5923

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 3014 | 0.00 | 

Pipeline 13020

File: doc/userguide/configuration/suricata-yaml.rst [hunk 1]
```diff
@@ -2589,7 +2604,7 @@ The Teredo decoder can be disabled. It is enabled by default.
         ports: $TEREDO_PORTS # syntax: '[3544, 1234]'
 
 Using this default configuration, Teredo detection will run on UDP port
-3544. If the `ports` parameter is missing, or set to `any`, all ports will be
+1.    If the `ports` parameter is missing, or set to `any`, all ports will be
```
Comment (victorjulien) [inline doc/userguide/configuration/suricata-yaml.rst:2607]: what is this?

File: doc/userguide/configuration/suricata-yaml.rst [hunk 2]
```diff
@@ -2035,32 +2033,49 @@ If the Suricata deployment is using multiple NICs on different NUMA nodes then h
...
+DPDK is natively supported by Mellanox and thus their NICs should work "out of the box".
 
-Although the capture interface uses DPDK library, there is no need to configure any lcores.
-The capture interface uses the standard Suricata threading module.
-Additionally, Suricata is intended to run as a primary process only.
+**Current DPDK support** involves Suricata running on:
+
+* a physical machine with a physical NICs such as:
+    * mlx5 (ConnectX-4/ConnectX-5/ConnectX-6)
+        * MCX416A-CCAT
```
Comment (victorjulien) [inline doc/userguide/configuration/suricata-yaml.rst:2052]: what is the purpose of listing these precise nics? think mentioning the higher level products (cx4/5/6, x(l)710, e810) is enough

File: doc/userguide/configuration/suricata-yaml.rst [hunk 3]
```diff
@@ -1975,30 +1975,30 @@ The whole DPDK configuration resides in the `dpdk:` node. This node encapsulates
           copy-iface: none # or PCIe address of the second interface
 
 
-The node `dpdk.eal-params` consists of `DPDK arguments <https://doc.dpdk.org/guides/linux_gsg/linux_eal_parameters.html>`_
-that are usually passed through command line. These arguments are used to initialize and configure EAL.
-Arguments can be specified in either long or short forms. When specifying the arguments, the dashes are omitted.
-Among other settings, this configuration node is able to configure available NICs to Suricata, memory settings or other
-parameters related to EAL.
+The `DPDK arguments <https://doc.dpdk.org/guides/linux_gsg/linux_eal_parameters.html>`_, which are typically provided through the command line, are contained in the node `dpdk.eal-params`.
```
Comment (victorjulien) [inline doc/userguide/configuration/suricata-yaml.rst:1978]: lets wrap the lines for better review reading

File: doc/userguide/configuration/suricata-yaml.rst [hunk 4]
```diff
@@ -2010,19 +2010,17 @@ The NUMA node to which the NIC is connected to can be determined from the file `
...
-Therefore, it is recommended to first find out to which NUMA node the NIC is connected to and only then allocate hugepages and set CPU cores affinity to the given NUMA node.
-If the Suricata deployment is using multiple NICs on different NUMA nodes then hugepages must be allocated on all of those NUMA nodes.
-
+Before Suricata can be run, it is required to allocate a sufficient number of hugepages. 
+For efficiency, hugepages are continuous chunks of memory (pages) that are larger (2 MB+) than what is typically used in the operating systems (4 KB).
+A lower count of pages allows faster lookup of page entries. The hugepages need to be allocated on the NUMA node where the NIC and affiniated CPU cores reside.
+For example, if the hugepages are allocated only on NUMA node 0 and the NIC is connected to NUMA node 1, then the application will fail to start.
+As a result, it is advised to identify the NUMA node to which the NIC is attached before allocating hugepages and setting CPU core affinity to that node.
+In case Suricata deployment uses multiple NICs, hugepages must be allocated on each of the NUMA nodes used by the Suricata deployment.
```
Comment (victorjulien) [inline doc/userguide/configuration/suricata-yaml.rst:2023]: below this it still suggests 8k 2M pages, which I think is too many. The default config will only use a few hundred in my testing, yet 16G is reserved.

Comment (victorjulien) [reply doc/userguide/configuration/suricata-yaml.rst:2023]: Below you have a calculation for how much memory a config requires, I wonder if we could read the available huge pages and then calculate if the config will fit? Then we could give a nice error/warning

Comment (lukashino) [reply doc/userguide/configuration/suricata-yaml.rst:2023]: wrt 8k 2M pages - wanted to change that as well but forgot about it. Thanks.

wrt hugepage calculation - I can look into it. I'll create a ticket.

Comment (lukashino) [reply doc/userguide/configuration/suricata-yaml.rst:2023]: https://redmine.openinfosecfoundation.org/issues/5966

File: doc/userguide/configuration/suricata-yaml.rst (hunk 3 above)
Comment (lukashino) [reply doc/userguide/configuration/suricata-yaml.rst:1978]: done :scissors:

**Comment by lukashino** [comment]
Continues in #8685

File: doc/userguide/configuration/suricata-yaml.rst (hunk 2 above)
Comment (victorjulien) [reply doc/userguide/configuration/suricata-yaml.rst:2052]: Seems like something worth documenting, but not in a user guide but rather in something like release notes

**Pull request closed without merging**